flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -2121,15 +2121,18 @@ void OmpAttributeVisitor::CreateImplicitSymbols(
       }
       return lastDeclSymbol;
     };
-    auto makeSharedSymbol = [&]() {
+    auto makeSharedSymbol = [&](bool setFlag) {
```

Comment:
Is this a bug:
```
    auto makePrivateSymbol = [&](Symbol::Flag flag) {
      const Symbol *hostSymbol =
          lastDeclSymbol ? lastDeclSymbol : &symbol->GetUltimate();
      lastDeclSymbol = DeclareNewPrivateAccessEntity(
          *hostSymbol, flag, context_.FindScope(dirContext.directiveSource));
      if (setFlag) {
        lastDeclSymbol->set(*setFlag);
      }
      return lastDeclSymbol;
    };
```
It has an argument `flag`, and then uses the outer scope `setFlag` to set on the created symbol? Seems like it doesn't need `flag` - or should be using `flag` instead of `setFlag`... :)

Reply:
Oh, no, it uses flag in a different place. Doh! :)